components/dropdown/dropdown-content-mixin.js

```diff
@@ -662,25 +710,64 @@ export const DropdownContentMixin = superclass => class extends RtlMixin(supercl
 			}
 		}
 
+		const specialMobileStyle = mediaQueryList.matches && this.mobileTray;
+		const mobileTrayRightLeft = mediaQueryList.matches && (this.mobileTray === 'right' || this.mobileTray === 'left');
```

Comment:
Lots of overrides happening here. Luckily I was able to use the same logic for left and right trays. For left and right trays;
-minimum width must be 285px 
-max width is 420px OR the total width - 30px for the backdrop
-width of the dropdown is set to the maximum width unless previously let by some other logic (but it must be within mix-max-widths.